Mathlib/CategoryTheory/Limits/IsLimit.lean

```diff
@@ -0,0 +1,1085 @@
+
+section Equivalence
+
+open CategoryTheory.Equivalence
+
+/-- If `s : Cone F` is a limit cone, so is `s` whiskered by an equivalence `e`.
+-/
+def whiskerEquivalence {s : Cone F} (P : IsLimit s) (e : K ≌ J) : IsLimit (s.whisker e.functor) :=
+  ofRightAdjoint (Cones.whiskeringEquivalence e).functor P
+#align category_theory.limits.is_limit.whisker_equivalence CategoryTheory.Limits.IsLimit.whiskerEquivalence
+
+/-- If `s : Cone F` whiskered by an equivalence `e` is a limit cone, so is `s`.
+-/
+def ofWhiskerEquivalence {s : Cone F} (e : K ≌ J) (P : IsLimit (s.whisker e.functor)) : IsLimit s :=
+  equivIsoLimit ((Cones.whiskeringEquivalence e).unitIso.app s).symm
+    (ofRightAdjoint (Cones.whiskeringEquivalence e).inverse P : _)
+#align category_theory.limits.is_limit.of_whisker_equivalence CategoryTheory.Limits.IsLimit.ofWhiskerEquivalence
+
+/-- Given an equivalence of diagrams `e`, `s` is a limit cone iff `s.whisker e.functor` is.
+-/
+def whiskerEquivalenceEquiv {s : Cone F} (e : K ≌ J) : IsLimit s ≃ IsLimit (s.whisker e.functor) :=
+  ⟨fun h => h.whiskerEquivalence e, ofWhiskerEquivalence e, by aesop_cat, by aesop_cat⟩
+#align category_theory.limits.is_limit.whisker_equivalence_equiv CategoryTheory.Limits.IsLimit.whiskerEquivalenceEquiv
+
+/-- We can prove two cone points `(s : Cone F).pt` and `(t.Cone G).pt` are isomorphic if
+* both cones are limit cones
+* their indexing categories are equivalent via some `e : J ≌ K`,
+* the triangle of functors commutes up to a natural isomorphism: `e.functor ⋙ G ≅ F`.
+
+This is the most general form of uniqueness of cone points,
+allowing relabelling of both the indexing category (up to equivalence)
+and the functor (up to natural isomorphism).
+-/
+@[simps]
+def conePointsIsoOfEquivalence {F : J ⥤ C} {s : Cone F} {G : K ⥤ C} {t : Cone G} (P : IsLimit s)
+    (Q : IsLimit t) (e : J ≌ K) (w : e.functor ⋙ G ≅ F) : s.pt ≅ t.pt :=
+  let w' : e.inverse ⋙ F ≅ G := (isoWhiskerLeft e.inverse w).symm ≪≫ invFunIdAssoc e G
+  { hom := Q.lift ((Cones.equivalenceOfReindexing e.symm w').functor.obj s)
+    inv := P.lift ((Cones.equivalenceOfReindexing e w).functor.obj t)
+    hom_inv_id := by
+      apply hom_ext P; intro j
+      dsimp
+      simp only [Limits.Cone.whisker_π, Limits.Cones.postcompose_obj_π, fac, whiskerLeft_app,
+        assoc, id_comp, invFunIdAssoc_hom_app, fac_assoc, NatTrans.comp_app]
+      rw [counit_app_functor, ←Functor.comp_map]
+      have l : 
+        NatTrans.app w.hom j = NatTrans.app w.hom (Prefunctor.obj (𝟭 J).toPrefunctor j) := by dsimp
+      rw [l,w.hom.naturality]
+      simp
+    inv_hom_id := by
+      apply hom_ext Q
+      aesop_cat }
+#align category_theory.limits.is_limit.cone_points_iso_of_equivalence CategoryTheory.Limits.IsLimit.conePointsIsoOfEquivalence
+
+end Equivalence
+
+/-- The universal property of a limit cone: a map `W ⟶ X` is the same as
+  a cone on `F` with vertex `W`. -/
```

Comment:
```suggestion
  a cone on `F` with cone point `W`. -/
```
... similarly in a few other places in this file.

Reply:
I furthered the propaganda.